src/core/providers/qgsabstractdatabaseproviderconnection.h

```diff
@@ -602,6 +651,14 @@ class CORE_EXPORT QgsAbstractDatabaseProviderConnection : public QgsAbstractProv
      */
     virtual QList<QList<QVariant>> executeSql( const QString &sql, QgsFeedback *feedback = nullptr ) const SIP_THROW( QgsProviderConnectionException );
 
+    /**
+     * Creates and returns a (possibly invalid) vector layer based on the \a sql statement and optional \a options.
+     * Raises a QgsProviderConnectionException if any errors are encountered or if SQL layer creation is not supported.
+     * \throws QgsProviderConnectionException
```

Comment:
```suggestion
     * \throws QgsProviderConnectionException if any errors are encountered or if SQL layer creation is not supported.
```

(results better presentation in the PyQGIS docs)